compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -1665,6 +1671,17 @@ object Scanners {
 
     def < (that: IndentWidth): Boolean = this <= that && !(that <= this)
 
+    /** Does `this` differ from `that` by not more than a single space? */
+    def isClose(that: IndentWidth): Boolean = this match
```

Comment:
perhaps `@tailrec`?

Reply:
Technically this is tail recursive. However we have the policy in the compiler that we will add @tailrec only for methods that could in practice recurse deeply if they would not be tail recursive. E.g. a method recursing over statement sequences, since a statement sequence can have lots of elements. Here, this is not really the case. There would be very few changes from spaces to tabs in leading whitespace, if any at all.

This policy is to avoid clutter and pointless busywork.